### PR TITLE
Make utf16_iter/utf8_iter deps optional in normalizer

### LINKED ISSUE #4988: Feature-gate UTF8 and UTF16 normalization
It reduces dependencies and compilation time.

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #4988

### CODE REVIEW

**Comment by Manishearth** [comment]
Questions:

 - Should these be default features?
 - The name we suggested in the meeting was after the crates. Should we instead just have `utf16` and `utf8_bytes`?

**Review by sffc — COMMENTED** [review]
Nit: There should be `✨ *Enabled with the `utf8_iter` Cargo feature.*` somewhere

**Comment by Manishearth** [comment]
I can't wait for `doc(cfg(auto))` to become a thing

File: components/normalizer/src/lib.rs [hunk 1]
```diff
@@ -1469,19 +1470,25 @@ macro_rules! normalizer_methods {
         ///
         /// Unpaired surrogates are mapped to the REPLACEMENT CHARACTER
         /// before normalizing.
+        ///
+        /// ✨ *Enabled with the `utf16_iter` Cargo feature.
```
Comment (sffc) [inline components/normalizer/src/lib.rs:1474]: Nit, here and elsewhere: missing the closing `*`

**Review by sffc — APPROVED** [review]

**Review by hsivonen — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**